<launch>
  <arg name="cam3_present"		default="true"/>

  <arg name="serial_no_camera1"    			default="937622073146"/> 			<!-- Note: Replace with actual serial number -->
  <arg name="serial_no_camera2"    			default="838212073567"/> 			<!-- Note: Replace with actual serial number -->
  <arg name="serial_no_camera3"    			default="920312073127"/> 			<!-- Note: Replace with actual serial number -->
  <arg name="serial_no_camera4"    			default="838212073332"/> 			<!-- Note: Replace with actual serial number -->
  <arg name="camera1"              			default="cam1"/>		<!-- Note: Replace with camera name -->
  <arg name="camera2"              			default="cam2"/>		<!-- Note: Replace with camera name -->
  <arg name="camera3"              			default="cam3"/>		<!-- Note: Replace with camera name -->
  <arg name="camera4"              			default="cam4"/>		<!-- Note: Replace with camera name -->

  <arg name="tf_prefix_camera1"         default="$(arg camera1)"/>
  <arg name="tf_prefix_camera2"         default="$(arg camera2)"/>
  <arg name="tf_prefix_camera3"         default="$(arg camera3)"/>
  <arg name="tf_prefix_camera4"         default="$(arg camera4)"/>
  <arg name="initial_reset"             default="false"/>
  <arg name="filters"			default="pointcloud"/>

<!-- launch the cameras -->


<group ns="$(arg camera2)">
 <include file="$(find realsense2_camera)/launch/includes/nodelet.launch.xml">
      
      <arg name="serial_no"             value="$(arg serial_no_camera2)"/>
      <arg name="tf_prefix"		value="$(arg tf_prefix_camera2)"/>
      <arg name="initial_reset"         value="$(arg initial_reset)"/>
      <arg name="filters"		value="$(arg filters)"/>
    </include>
  </group>
    
 <group ns="$(arg camera1)"> 
    <include file="$(find realsense2_camera)/launch/includes/nodelet.launch.xml">
      
      <arg name="serial_no"             value="$(arg serial_no_camera1)"/>
      <arg name="tf_prefix"         		value="$(arg tf_prefix_camera1)"/>
      <arg name="initial_reset"         value="$(arg initial_reset)"/>
      <arg name="filters"		value="$(arg filters)"/>
    </include>
 </group> 

<group ns="$(arg camera4)"> 
    <include file="$(find realsense2_camera)/launch/includes/nodelet.launch.xml">
      
      <arg name="serial_no"             value="$(arg serial_no_camera4)"/>
      <arg name="tf_prefix"         		value="$(arg tf_prefix_camera4)"/>
      <arg name="initial_reset"         value="$(arg initial_reset)"/>
      <arg name="filters"		value="$(arg filters)"/>
    </include>
 </group> 
  
  
<!-- launch camera 3 with OpenNI body tracker -->

<include file="$(find cob_openni2_tracker)/launch/body_tracker_nodelet.launch" if="$(eval cam3_present != '')">
      
      <arg name="standalone_without_camera_driver"   value="true"/>
	<arg name="camera_name"   value="$(arg camera3)"/>
	
</include>

<!-- launch the tf publishers -->

<node name="tf_base_link" pkg="tf" type="static_transform_publisher" output="screen" args="0 0 0 0 0 0 world base_link 50" />

<node name="tf_cam1_link" pkg="tf" type="static_transform_publisher" output="screen" args="0.458 1.1475 1.072 -2.0944 0.7854 -1.5708 base_link cam1_link 50" />

<node name="tf_cam2_link" pkg="tf" type="static_transform_publisher" output="screen" args="-0.418 1.11 1.072 -1.0472 0.7854 -1.5708 base_link cam2_link 50" />

<node name="tf_cam3_link" pkg="tf" type="static_transform_publisher" output="screen" args="-0.45 0.65 0.18 -1.5708 0 -1.5708 base_link cam3_link 50"  if="$(eval cam3_present != '')"/>

<node name="tf_cam4_link" pkg="tf" type="static_transform_publisher" output="screen" args="0.408 -0.235 1.097 2.0944 0.7854 -1.5708 base_link cam4_link 50" />

<!-- launch the robot -->

<include file="$(find ur_manipulation)/launch/ur5e_bringup.launch" />

<!-- launch the robot pose publisher -->

<include file="$(find seher_support)/launch/links_pose_publisher.launch" />

<!-- launch Moveit -->

<include file="$(find ur5e_egp50_moveit_config)/launch/ur5e_egp50_moveit_planning_execution.launch" />


<!-- launch rviz -->

<node name="rviz" pkg="rviz" type="rviz" args="-d $(find seher)/launch/rviz_config_workcell.rviz" />

<!-- launch the obstacle publisher -->

<node name="obstacle_publisher" pkg="seher" type="collision_objects" />

<!-- launch the pointcloud computing node -->

<node name="pointcloud_computing" pkg="seher" type="pointcloud_computing" />

<!-- launch the robot self filter -->

<include file="$(find seher)/launch/robot_self_filter.launch" />

<!-- launch the statistical outlier removal filter -->


<node name="pointcloud_statistical_removal_filter" pkg="seher" type="pointcloud_statistical_outlier_removal_filter" />

<!-- launch the distance computing node -->


<node name="distance_computing_node" pkg="seher" type="distance_calculation" />

</launch>
